mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -449,6 +449,16 @@ class SparseVector(
   override def toString: String =
     "(%s,%s,%s)".format(size, indices.mkString("[", ",", "]"), values.mkString("[", ",", "]"))
 
+  override def equals(other: Any): Boolean = {
+    other match {
+      case v: SparseVector =>
+        this.size == v.size &&
+          util.Arrays.equals(this.indices, v.indices) &&
+          util.Arrays.equals(this.values, v.values)
```

Comment:
Nice suggestion. This should be your contribution and I'll close the PR. Thanks.